wordpress-mobile/WordPress-Android
# Issue/10649 material reader post list

The first part of #10649

This PR moves Reader Post List and Tags & Sites to Material Theme and adds Dark Theme support. There are some code changes for cosmetic reasons. 

@develric @osullivanchris I would like to loop you guys in since this PR touches Subfilter, which is part of IA.

Reader without IA changes:

 ![comparison_reader_no_ia](https://user-images.githubusercontent.com/728822/73114073-74ad8200-3ecc-11ea-8826-c2bdc70accf9.png)

With IA changes (alpha branch):

![comparison_reader_post_list](https://user-images.githubusercontent.com/728822/73114140-db32a000-3ecc-11ea-95d7-338491a4dde1.jpg)


To test:
- Navigate to Reader post list, and switch between different filters/subfilters. Make sure everything looks good in both light and dark themes.
- Navigate to Reader's Tags & Sites screen, and make sure everything looks good in both light and dark themes.

You can switch the theme from App Settings.

PR submission checklist:
